cardano-prelude/src/Cardano/Prelude/Base.hs

```diff
@@ -1,62 +1,269 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE ExplicitNamespaces #-}
 {-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE Safe #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE Trustworthy #-}
 
 module Cardano.Prelude.Base (
   module X,
+  HasLength (..),
+  LByteString,
+  LText,
+  head,
   identity,
-  putTextLn,
   length,
-#if __GLASGOW_HASKELL__ >= 906
-  type (~)
-#endif
+  liftM',
+  liftM2',
+  map,
+  putTextLn,
+  scanl',
+  throwIO,
+  trace,
+  traceM,
+  traceShow,
 )
 where
 
-import Protolude as X hiding (
-  Hashable,
-  Map,
-  hash,
-  hashUsing,
-  hashWithSalt,
-  identity,
-  length,
-  witness,
-  (.),
- )
-import qualified Protolude as Y
-
 import Data.Map.Strict as X (Map)
-import qualified Data.Text as T
+import Data.List (scanl')
+import Data.Text qualified as Text
+import Data.Text.IO qualified as Text
+import Prelude qualified
 
-import Control.Category (id)
+import Control.Category qualified as Category
 import Control.Category as X hiding (id)
 import Numeric.Natural as X
+import Control.Applicative as X (Applicative (..), Alternative (..), Const (..), ZipList (..),
+        (<**>), liftA, liftA2, liftA3, optional, pure)
+import Control.Concurrent as X (ThreadId, forkIO, forkFinally, forkIOWithUnmask,
+        killThread, forkOn, forkOnWithUnmask, getNumCapabilities, setNumCapabilities,
+        threadCapability, yield, threadDelay, threadWaitRead, threadWaitWrite,
+#if MIN_VERSION_base(4,7,0)
```

Comment:
We should really write down which GHC releases these are. Going from indirect base version -> figure out which GHC release, is always so annoying.

Reply:
I always refer to https://www.snoyman.com/base/

Reply:
This is my favorite table when it comes to all of the core packages that are wired in with ghc:

https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history

Reply:
That's a good one too @lehins! The annoying part is, that a lot of those don't necessarily need to be wired in :-( 
The trend of shipping more and more packages with GHC, instead of less is worrying to me. Even worse, Cabal maintains its own list of non-reinstallable packages.